Plugins/SwiftGRPCPlugin/plugin.swift

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2022, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import PackagePlugin
+
+@main
+struct SwiftGRPCPlugin: BuildToolPlugin {
+  /// Errors thrown by the `SwiftGRPCPlugin`
+  enum PluginError: Error {
+    /// Indicates that the target where the plugin was applied to was not `SourceModuleTarget`.
+    case invalidTarget
+    /// Indicates that the file extension of an input file was not `.proto`.
+    case invalidInputFileExtension
+  }
+
+  /// The configuration of the plugin.
+  struct Configuration: Codable {
+    /// Encapsulates a single invocation of protoc.
+    struct Invocation: Codable {
+      /// The visibility of the generated files.
+      enum Visibility: String, Codable {
+        /// The generated files should have `internal` access level.
+        case `internal`
+        /// The generated files should have `public` access level.
+        case `public`
+      }
+
+      /// An array of paths to `.proto` files for this invocation.
+      var protoFiles: [String]
+      /// The visibility of the generated files.
+      var visibility: Visibility?
+      /// Whether server code is generated.
+      var server: Bool?
+      /// Whether client code is generated.
+      var client: Bool?
+      /// Whether test client code is generated.
+      var testClient: Bool?
```

Comment:
This is deprecated so we can leave out the option (until the time that someone complains...)